src/niscope/templates/session.py/fancy_fetch_array_measurement.py.mako

```diff
@@ -17,9 +17,6 @@
             self._fetch_num_records = -1 if num_records is None else num_records
             self._fetch_meas_num_samples = -1 if meas_num_samples is None else meas_num_samples
 
-        if meas_wfm_size is None:
-            meas_wfm_size = self._actual_meas_wfm_size(array_meas_function)
-
```

Comment:
I wonder if it might be worth having extra-fancy code here, e.g.
```python
if isinstance(self._interpreter, _library_interpreter.LibraryInterpreter):
    if meas_wfm_size is None:
        meas_wfm_size = self._actual_meas_wfm_size(array_meas_function)
else:
    if measurement_waveform_size is not None:
        raise ValueError('The argument "meas_wfm_size" should be None when using gRPC.')
```
... so that we can avoid having to copy the generated Interpreter code and make just a slight modification.
@marcoskirsch ?

Reply:
Yeah, it feels like handling this lies a bit outside of the interpreter's responsibilities. Whether we shove it into the interpreter or not, the solution won't be pretty. I feel like David presents a solid case for handling this in session.py, though.

Reply:
I'm ok with @DavidCurtiss 's suggestion

Reply:
Yeah, that sounds good. Switched to put David's suggested code in there and simplify the generation for the interpreters.